### DISCUSSION TITLE
Does Uno provide APIs for events like: keypress, clipboard, mouse select...

### DISCUSSION THREAD

**Discussion by robertpatan** [unoplatform/uno#11822] [Q&A]
Hi,
Does Uno provide APIs for the following events:
 - keypress and capture the value
- clipboard
- mouse clicks
- mouse select of a text and capture the value
-  to be able to detect a focus in a input/textarea from other applications or either a browser
- possibility to run in the background as a service

Basically I want to make an application similar with Grammarly for Windows/Mac.

Would it be possible? If yes, would it be more difficult then making it the native platform itself?

Thanks!

**Comment by jeromelaban** [comment]
Hi @robertpatan, thanks for the question.

Uno Platform's own APIs (mirrored from WinAppSDK) in particular don't provide cross-process features that we know of. 

That being said, the underlying platform is [readily available to use](https://platform.uno/docs/articles/platform-specific-csharp.html), whether it is mac or Windows. Win32 is definitely useable on Windows.

For macOS, our current target uses `net7.0-maccatalyst` (as there are [stability issues](https://github.com/unoplatform/uno/issues/982) for `net7.0-macos`) and you may be restricted by the app model sandbox. You'll need to take a look at the Catalyst APIs to determine if such interactions are possible whether it's an app installed from the app store, or locally by the user.
